### PR TITLE
fix: start the BTC adapter only by accepting a config

### PR DESCRIPTION
**Pull request by rumenov**
To enable the addition of the BTC adapter to pocket ic

### CODE REVIEW

File: rs/bitcoin/adapter/src/rpc_server.rs [hunk 1]
```diff
@@ -136,26 +136,29 @@ pub fn start_grpc_server(
...
+            let incoming = incoming_from_path(uds_path);
+            let server_fut = Server::builder()
+                .add_service(BtcServiceServer::new(btc_adapter_impl))
+                .serve_with_incoming(incoming);
+            tokio::spawn(async move {
+                server_fut.await.expect("gRPC server crashed");
+            });
+        }
+        IncomingSource::Systemd => {
+            let incoming = unsafe { incoming_from_first_systemd_socket() };
```
Comment (tthebst) [inline rs/bitcoin/adapter/src/rpc_server.rs:151]: Doesn't this panic if it is not within a tokio runtime?

**Review by tthebst — APPROVED** [review]

File: rs/bitcoin/adapter/src/lib.rs [hunk 2]
```diff
@@ -49,14 +51,12 @@ mod transaction_store;
 // malicious fork can be prioritized by a DFS, thus potentially ignoring honest forks).
 mod get_successors_handler;
 
-pub use blockchainmanager::BlockchainManager;
+use crate::{
+    common::BlockHeight, config::IncomingSource, router::start_main_event_loop, stream::StreamEvent,
+};
 pub use blockchainstate::BlockchainState;
```
Comment (mraszyk) [inline rs/bitcoin/adapter/src/lib.rs:57]: Don't you also want to hide this?

File: rs/bitcoin/adapter/src/lib.rs [hunk 3]
```diff
@@ -193,16 +193,31 @@ impl AdapterState {
...
+    let _enter = rt_handle.enter();
+
+    // Metrics server should only be started if we are managed by systemd and receive the
+    // metrics socket as FD(4).
+    // SAFETY: The process is managed by systemd and is configured to start with at metrics socket.
+    // Additionally this function is only called once here.
+    // Systemd Socket config: ic-https-outcalls-adapter.socket
+    // Systemd Service config: ic-https-outcalls-adapter.service
+    if config.incoming_source == IncomingSource::Systemd {
+        let stream = unsafe { incoming_from_nth_systemd_socket(2) };
```
Comment (mraszyk) [inline rs/bitcoin/adapter/src/lib.rs:211]: in the comment, you refer to FD(4)

File: rs/bitcoin/adapter/src/lib.rs [hunk 4]
```diff
@@ -193,16 +193,31 @@ impl AdapterState {
 }
 
 /// Starts the gRPC server and the router for handling incoming requests.
-pub fn start_grpc_server_and_router(
-    config: &config::Config,
+pub fn start_server(
```
Comment (mraszyk) [inline rs/bitcoin/adapter/src/lib.rs:196]: Do you want me to switch to this function in my PocketIC PR?

File: rs/bitcoin/adapter/src/lib.rs (hunk 2 above)
Comment (rumenov) [reply rs/bitcoin/adapter/src/lib.rs:57]: It is used in the benches, will do it in a followup PR

File: rs/bitcoin/adapter/src/lib.rs (hunk 4 above)
Comment (rumenov) [reply rs/bitcoin/adapter/src/lib.rs:196]: exactly

File: rs/bitcoin/adapter/src/rpc_server.rs (hunk 1 above)
Comment (rumenov) [reply rs/bitcoin/adapter/src/rpc_server.rs:151]: start_grpc_server is always called within a runtime, I know it is implicit most likely I will fix it in followup PRs

File: rs/bitcoin/adapter/src/lib.rs (hunk 3 above)
Comment (rumenov) [reply rs/bitcoin/adapter/src/lib.rs:211]: I believe the comment is not for the BTC adapter but for the socket that the adapter metrics use

**Pull request merged**